### ISSUE TITLE
[LiveComponent] v2.9 Autocomplete field not refreshed

### ISSUE THREAD

**Issue by bastien70** [symfony/ux#909]
I am currently on version 2.9 for Live-component, Twig-component, and Autocomplete.
We updated from 2.7, and we ran `yarn install --force` and `yarn run dev`. We also updated Autocomplete and Live-component recipes.

In version 2.7, when we had an AutocompleteType field (from Symfony UX), and an Ajax request goes, the Autocomplete field kept its "TomSelect" well.

Oddly since version 2.9, this one loses its "TomSelect". However, when we retrieve the TomSelect instance of the Autocomplete field, it seems well initialized and has the right parameters.

https://github.com/symfony/ux/assets/53140475/c5e21a21-57c9-41e2-afc2-5b5a3a527307

Before first Ajax request :
![1](https://github.com/symfony/ux/assets/53140475/76ec4b83-d335-451b-bcbe-036f9869422c)

After first Ajax request : 
![2](https://github.com/symfony/ux/assets/53140475/228973b1-39ad-4ee0-a06a-d4a74335dd60)

And TomSelect attributes at this moment : 

![3](https://github.com/symfony/ux/assets/53140475/b0100031-779a-4094-beff-35e7f58a5f31)

In our form, we have : 

```php
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $builder
            ->add('course', ListingSchoolClassCourseAutocompleteField::class, [
                'label' => 'Course',
                'required' => true,
            ])
        ;
        
        // ...
}
```

And this is our ListingSchoolClassCourseAutocompleteField : 

```php
#[AsEntityAutocompleteField(alias: 'listing_school_class_course', route: 'ux_entity_autocomplete')]
class ListingSchoolClassCourseAutocompleteField extends AbstractType
{
    public function __construct(
        private readonly Security $security,
        private readonly SessionService $sessionService,
        private readonly ScopeService $scopeService
    ) {
    }

    public function configureOptions(OptionsResolver $resolver)
    {
        /** @var User $user */
        $user = $this->security->getUser();
        $collegeYear = $this->sessionService->getCollegeYear();

        $resolver->setDefaults([
            'class' => SchoolClassCourse::class,
            'placeholder' => '',
            'choice_label' => 'createShortLabel',
            'no_more_results_text' => false,
            'query_builder' => function () use ($user, $collegeYear) {
                return $this->scopeService->getSchoolClassCourses($user, $collegeYear, SchoolClass::STATUS_INSCRIPTION, true);
            },
            'searchable_fields' => [
                'schoolClass.shortName',
                'course.yearInCurriculum',
                'course.shortName',
            ],
        ]);
    }

    public function getParent(): string
    {
        return ParentEntityAutocompleteType::class;
    }
}

```

Is it a bug ? Or a mistake?
Thanks!

**Comment by weaverryan** [comment]
@bastien70 Can you check this on 2.8.1 also? My guess is that 2.8 is where this behavior changed, but I'd like to verify. We actually added code IN 2.8 to help this situation... so it's interesting that it would be a step backwards. I don't think we've gotten any other reports of problems, but 2.8 does contain some pretty serious code to get TomSelect & LiveComponents to play together nicely (necessary because TomSelect is a bit weird about updating itself when its underlying options change). In 2.7, these played together mostly nicely because we added a `data-live-ignore`, but that had limitations in some more advanced cases (like if you need to change the options in your select at runtime).

**Comment by bastien70** [comment]
Hey @weaverryan I just tested on 2.8.1. Same thing
We were actually able to circumvent this with a data-live-ignore, but indeed, for fields that will have their options modified, it will unfortunately not be able to work.

From memory I had the same problem already from version 2.8.0. Maybe something has changed from this version?

**Comment by weaverryan** [comment]
This may now be fixed in #1290. Please reopen if you notice the problems after the next release.

Cheers!
